Platon/backend/app/workspace_system/views.py

```diff
@@ -645,6 +668,209 @@ def delete(user_id, self):
+    @active_contribution_required(param_loc='form',workspace_id_key='workspace_id')
+    def post(user_id, self):
+        '''
+            Creates milestone
+        '''
+        form = PostMilestoneForm(request.form)
+        if form.validate():
+            
+            try: 
+                # Create milestone record
+                milestone = Milestone(user_id, form.workspace_id.data, form.title.data, form.description.data, form.deadline.data)
+            except:
+                return make_response(jsonify({'error': 'milestone record is not created'}), 500)
+
+            try:
+                db.session.add(milestone)
+                db.session.commit()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            return make_response(jsonify({'msg': 'milestone is successfully created'}), 200)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error', 404: 'Not found'})
+    @api.expect(put_milestone_parser)
+    @login_required
+    @workspace_exists(param_loc='form',workspace_id_key='workspace_id')
+    @active_contribution_required(param_loc='form',workspace_id_key='workspace_id')
+    def put(user_id, self):
+        '''
+            Updates a milestone
+        '''
+        form = PutMilestoneForm(request.form)
+        if form.validate():
+            
+            try:
+                # Gets the parameters from the form data.
+                new_attributes = {}
+                for key, value in form.data.items():
+                    if value or (value == 0):
+                        # these attributes shouldn't be changed
+                        if key == 'milestone_id' or key == 'workspace_id':
+                            continue
+                        new_attributes[key] = value
+            except:
+                return make_response(jsonify({'error': 'Form data usage is incorrect'}), 500)
+
+            try:
+                milestone = Milestone.query.filter(Milestone.id == form.milestone_id.data).first()
+
+                if milestone is None:
+                    return make_response(jsonify({'error': 'milestone not found'}), 404)
+
+                if new_attributes:
+                    # Updates the attributes of the milestone in the database.
+                    for key, value in new_attributes.items():
+                        if key == 'title':
+                            milestone.title = value
+                        elif key == 'description':
+                            milestone.description = value
+                        elif key == 'deadline':
+                            milestone.deadline = value
+                        else:
+                            return make_response(jsonify({'error': 'Attribute name {} is incorrect'.format(key)}), 500)
```

Comment:
"form.validate()" already does this key name validation, am I wrong? So, is this line that raises an error necessary?

Reply:
You're right. No, it is not necessary as you've mentioned. I've deleted this part with the latest changes.